app/javascript/components/widgets/guests/tab.jsx

```diff
@@ -121,7 +121,13 @@ const Guests = () => (
                     prev,
                   );
                 }
-                return prev;
+
+                const isFutureToday = item => isInFutureToday(item.publishOn);
```

Comment:
It seems like we could still have old announcements if we add a new announcement that is in the future. The solution might be chaining it all together: append new event, filter, sort.

Reply:
So I tested this and if a future announcement added and there is one in the list that is in the past then the filter will take care of it. The problem I see is that a past announcement will remain in the list until the next poll (which happens every hour) or until the next published announcement. So maybe the `pollInterval` could be shorter or we would be ok with having an event that happened an hour ago until the next poll.